src/MagnumPlugins/BasisImporter/BasisImporter.cpp

```diff
@@ -0,0 +1,251 @@
+/*
+    This file is part of Magnum.
+
+    Copyright © 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019
+              Vladimír Vondruš <mosra@centrum.cz>
+    Copyright © 2019 Jonathan Hale <squareys@googlemail.com>
+
+    Permission is hereby granted, free of charge, to any person obtaining a
+    copy of this software and associated documentation files (the "Software"),
+    to deal in the Software without restriction, including without limitation
+    the rights to use, copy, modify, merge, publish, distribute, sublicense,
+    and/or sell copies of the Software, and to permit persons to whom the
+    Software is furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included
+    in all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNETCION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+#include "BasisImporter.h"
+
+#include <basisu_transcoder.h>
+
+#include <Corrade/Containers/Optional.h>
+#include <Corrade/Containers/StaticArray.h>
+#include <Corrade/Utility/ConfigurationGroup.h>
+#include <Corrade/Utility/ConfigurationValue.h>
+#include <Corrade/Utility/Debug.h>
+#include <Corrade/Utility/String.h>
+#include <Magnum/PixelFormat.h>
+#include <Magnum/Trade/ImageData.h>
+
+namespace Corrade { namespace Utility {
+
+/* Configuration value implementation for BasisTranscodingType */
+template<> struct ConfigurationValue<Magnum::Trade::BasisTranscodingType> {
+
+    static std::string toString(
+        const Magnum::Trade::BasisTranscodingType& value,
+        ConfigurationValueFlags)
+    {
+        #define _c(value) case Magnum::Trade::BasisTranscodingType::value: return #value;
+        switch(value) {
+            _c(Etc1)
+            _c(Etc2)
+            _c(Bc1)
+            _c(Bc3)
+            _c(Bc4)
+            _c(Bc5)
+            _c(Bc7M6OpaqueOnly)
+            _c(Pvrtc1_4OpaqueOnly)
+        }
+        #undef _c
+
+        return "<invalid>";
+    }
+
+    static Magnum::Trade::BasisTranscodingType fromString(
+        const std::string& value, ConfigurationValueFlags)
+    {
+        const Containers::StaticArray<8, const char*> FormatNames{
```

Comment:
I don't think this needs a StaticArray. Range-based for will work with `const char* FormatNames[]` as well (remove the include, too). 

Actually I think it would be good to have it outside of the function and next to the `TranscoderTextureFormat`, because those two need to have the same order.

Reply:
This will create a couple more `namespace ... {` sections. I guess that's alright?

Reply:
Yeah that's fine. The way you just did that is fine as well.